statefun-flink/statefun-flink-launcher/src/main/java/org/apache/flink/statefun/flink/launcher/StatefulFunctionsJobGraphRetriever.java

```diff
@@ -113,4 +124,10 @@ private PackagedProgram createPackagedProgram() {
       throw new RuntimeException("Unable to construct a packaged program", e);
     }
   }
+
+  private static int resolveParallelism(int parallelism, Configuration configuration) {
```

Comment:
This seems a bit tricky,
Consider the case where a user set a value 3, in Flink-conf.yaml
But for testing purposes they provide -p with a value equal to PATALLEISM_DEFAULT.

With what degree of parallelism would the job start?

Reply:
The parallelism used will be the value provided via `-p`.

Note that `ExecutionConfig.PARALLELISM_DEFAULT` is `-1`. It's merely a flag to indicate that no parallelism was set.

Reply:
But I think you have a point -
We should probably when parsing the command line, make sure that the value provided via `-p` is `>= 1`. That would avoid the problem you mentioned.

Reply:
Aha! In that case -1 works!
Then consider this comment to be resolved.